test/ledger.js

```diff
@@ -0,0 +1,123 @@
+var async       = require("async");
+var assert      = require('assert');
+var lodash       = require('lodash');
+var Amount      = require("ripple-lib").Amount;
+var Remote      = require("ripple-lib").Remote;
+var Transaction = require("ripple-lib").Transaction;
+var Server      = require("./server").Server;
+var testutils   = require("./testutils");
+var config      = testutils.init_config();
+
+suite('Ledger requests', function() {
+  var $ = { };
+  var expectedledgers = [
+    {"ledger":{"accepted":true,"account_hash":"A21ED30C04C88046FC61DB9DC19375EEDBD365FD8C17286F27127DF804E9CAA6","closed":true,"ledger_index":"1","seqNum":"1","totalCoins":"100000000000000000","total_coins":"100000000000000000","transaction_hash":"0000000000000000000000000000000000000000000000000000000000000000"},"ledger_index":1,"validated":true},
+    {"ledger":{"accepted":true,"account_hash":"A21ED30C04C88046FC61DB9DC19375EEDBD365FD8C17286F27127DF804E9CAA6","close_time_resolution":30,"closed":true,"ledger_index":"2","seqNum":"2","totalCoins":"100000000000000000","total_coins":"100000000000000000","transaction_hash":"0000000000000000000000000000000000000000000000000000000000000000"},"ledger_index":2,"validated":true},
+    {"ledger":{"closed":false,"ledger_index":"3","seqNum":"3"},"ledger_current_index":3,"validated":false},
+    {"error":"remoteError","error_message":"Remote reported an error.","remote":{"error":"lgrNotFound","error_code":20,"error_message":"ledgerNotFound","id":4,"request":{"command":"ledger","id":4,"ledger_index":4},"status":"error","type":"response"}}
+  ];
+
+  var specialIndexToExpected = {
+    "validated": 1,
+    "closed": 1,
+    "current": 2
+  };
+
+  var filterFields = function (ledger) {
+    if(typeof ledger === 'object') {
+      ledger = lodash.omit(ledger, ['close_time', 'close_time_human', 'hash', 'ledger_hash', 'parent_hash']);
+      Object.keys(ledger).map(function(key) {
+        ledger[key] = filterFields(ledger[key]);
+      });
+    }
+    return ledger;
+  }
+
+  var goodLedger = function (index, self, callback) {
+    self.what = "Good Ledger " + index;
+
+    var request = $.remote.request_ledger(index)
+      .on('success', function(ledger) {
+        // console.log("Ledger: %s", JSON.stringify(ledger));
+        ledger = filterFields(ledger);
+        // console.log("Ledger: %s", JSON.stringify(ledger));
+
+        var expected = specialIndexToExpected[index];
+        if(typeof expected === 'undefined')
```

Comment:
There's not much difference, but `typeof` seems a bit safer, and is used several times elsewhere in the tests (though not to check for `undefined`). https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined